### ISSUE TITLE
How do I use a different python version for prompt flow custom runtime? 

### ISSUE THREAD

**Issue by sparse-coder** [microsoft/llmops-promptflow-template#154]
We are using our own environment definition to build the promptflow runtime image, and use it as the base for provisioning the custom runtime on AML compute instance for doing batch and evaluation runs.

So far we are good.

What we need is a way to use a different python version e.g. Python 3.11 on my custom runtime. Is there a way I can do this? So far what I have observed is all the base images supplied by mcr use python 3.9.   

My docker file looks like this
`FROM mcr.microsoft.com/azureml/promptflow/promptflow-runtime-stable:20240515.v1
COPY ./* ./
RUN pip install -r requirements.txt
`
Is there a way I can define something in my docker file to upgrade the python version?

**Comment by stewartadam** [comment]
We're looking to do something similar, you can use `conda` to update the runtime env:

```
FROM mcr.microsoft.com/azureml/promptflow/promptflow-runtime:latest

# Update to Python 3.11
RUN conda install -y 'python=3.11' -p /azureml-envs/prompt-flow/runtime
```

**Comment by sparse-coder** [comment]
@stewartadam Thank you! I did the same thing by updating the conda env, but this idea of copying the promptflow-runtime wheel didn't occur to me. Good idea👍 BTW!

For now we have decided to not go through this route!

Thank you!

**Comment by ebotiab** [comment]
Hi, is this issue expected to be fixed in the future?

**Comment by WilliamWatsonEAI** [comment]
@ebotiab agreed, this should not be so hard to change base image Python version

**Comment by lovedeepatsgit** [comment]
@stewartadam can you please also tell me how to adjust the liveness and readiness of the container, And what do you mean by that promptflow_runtime whl is not published. And, I have also taken reference from your suggestion, but at the end I got stuck on a point in the final deployment, below is the error I am facing:
![Image](https://github.com/user-attachments/assets/fdd8eb79-01d7-465e-b5b2-3bdc5a899a77)

**Comment by O-EAI** [comment]
Would be great to use python3.11
